gef-ui/server/controllers/amendments/submit-amendment-to-ukef/get-submit-amendment-to-ukef.ts

```
@@ -0,0 +1,50 @@
+import { CustomExpressRequest, PORTAL_AMENDMENT_STATUS } from '@ukef/dtfs2-common';
+import { Response } from 'express';
+import api from '../../../services/api';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+
+export type GetSubmitAmendmentToUkefRequest = CustomExpressRequest<{
+  params: { dealId: string; facilityId: string; amendmentId: string };
+}>;
+
+/**
+ * controller to get the submit to ukef confirmation page
+ *
+ * @param req - The express request
+ * @param res - The express response
+ */
+export const getSubmitAmendmentToUkef = async (req: GetSubmitAmendmentToUkefRequest, res: Response) => {
+  try {
+    const { dealId, facilityId, amendmentId } = req.params;
+    const { userToken } = asLoggedInUserSession(req.session);
+
+    const deal = await api.getApplication({ dealId, userToken });
+    const { details: facility } = await api.getFacility({ facilityId, userToken });
+
+    if (!deal || !facility) {
+      console.error('Deal %s or Facility %s was not found', dealId, facilityId);
+      return res.redirect('/not-found');
+    }
+
+    const amendment = await api.getAmendment({ facilityId, amendmentId, userToken });
+
+    if (!amendment) {
+      console.error('Amendment %s not found for the facility %s', amendmentId, facilityId);
+      return res.redirect('/not-found');
+    }
+
+    if (amendment.status !== PORTAL_AMENDMENT_STATUS.READY_FOR_CHECKERS_APPROVAL) {
```

Should we also add `Draft` and `Maker's inpur required` statues.

---

@abhi-markan sorry I don't understand your comment. This controller is to get the confirmation page, but confirmation page should appear to the user and they will be able to submit the amendment only in case the status is ready for checkers approval, right ?. If status is Draft or Maker's input or Acknowledged then should throw an error